### PR TITLE
Update to Libtorch v1.11.0

### LINKED ISSUE #805: LibTorch v1.11
https://github.com/pytorch/pytorch/releases/tag/v1.11.0

### PR DESCRIPTION
**Pull request by dfalbel**
Fix #805

### CODE REVIEW

File: .github/workflows/lantern.yaml [hunk 1]
```diff
@@ -6,7 +6,7 @@ on:
       - master
       - main
       - "cran/**"
-      - "libtorch1.10.2"
+      - "libtorch-v1.11.0"
```
Comment (hsbadr) [inline .github/workflows/lantern.yaml:9]: Need to update the default version and the docs:
```r
Warning: Codoc mismatches from documentation object 'get_install_libs_url':
get_install_libs_url
  Code: function(version = "1.11.0", type = install_type(version =
                 version))
  Docs: function(version = "1.10.2", type = install_type(version =
                 version))
```

File: R/linalg.R [hunk 2]
```diff
@@ -260,15 +262,17 @@ linalg_cond <- function(A, p = NULL) {
...
 #' @export
-linalg_matrix_rank <- function(A, tol = NULL, hermitian = FALSE) {
-  if (is.null(tol)) {
-    torch_linalg_matrix_rank(self = A, tol = tol, hermitian = hermitian)
-  } else {
-    if (!is_torch_tensor(tol)) {
-      tol <- torch_scalar_tensor(tol)
-    }
-    torch_linalg_matrix_rank(input = A, tol = tol, hermitian = hermitian)
+linalg_matrix_rank <- function(A, ..., atol = NULL, rtol = NULL, tol = NULL, hermitian = FALSE) {
```
Comment (hsbadr) [inline R/linalg.R:265]: Need to fix the following warning (define/document `...` argument):
```r
Warning:   Undocumented arguments in documentation object 'linalg_matrix_rank'
    ‘...’
```

**Comment by dfalbel** [comment]
Really struggling [with](https://github.com/mlverse/torch/runs/6614901627?check_suite_focus=true#step:8:34): 

> CMake Error at D:/a/torch/build/libtorch/share/cmake/Caffe2/public/cuda.cmake:41 (enable_language):
  No CMAKE_CUDA_COMPILER could be found.

on Windows.

**Comment by dfalbel** [comment]
Thanks @hsbadr !! That seems to work, at least no failures until now :)

**Pull request merged**